pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
+  all (`connect=True`) subsequently created :class:`~pywbem.WBEMConnection`
+  objects or for a particular connection (`connect=conn` where conn is a
+  WBEMConnection object).
+
+.. _`Logging configuration methods`:
+
+Logging configuration methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The pywbem loggers may be configured through the following pywbem methods and
+must be used to set the log detail_level and activate the loggers.
+
+  * :meth:`~pywbem.WBEMConnection.configure_logger` - Configures and
+    optionally activates one or more of the loggers that log user-issued calls
+    to :class:~pywbem.WBEMConnection methods that drive WBEM operations and
+    pywbem http requests and responses. Thus, for example, the following
+    would configure the api logger as a file logger for a single connection::
+
+       conn = WBEMConnection(...)
+       WBEMConnection.configure_logger('api', dest=file, log_filename='blah.log'
+                                       connection=conn)
+
+  * :func:`configure_loggers_from_string` - Allows configuring and
+    activating the pywbem loggers from a single formatted string. This is most
+    useful in defining the loggers from a command line tool such as pywbemcli.
+    Thus, for example, the following could be used to configure and
+    activate both loggers::
+
+        configure_loggers_from_string('api=file:summary,http=stderr:all',
+                                      log_filename='mylog.log',
+                                      connection='True')
+
+.. _`Logging configuration Examples`:
+
+Logging configuration examples
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Configure and activate loggers with
+:meth:`~pywbem.WBEMConnection.configure_logger`.
+
+* Configure all WBEMConnections api logger for summary information output to
+  a file::
+
+    WBEMConnection.configure_logger('api', dest='stderr',
+                                    detail_level='summary',
+                                    connection=True)
+    conn = WBEMConnection()  # logs WBEMConnection api calls and responses
+
+* Configure a single WBEMConnection for output of summary information::
+
+    conn = WBEMConnection()
+    WBEMConnection.configure_logger('all', dest='file', log_filname='xxx.log',
+                                        connection=conn)
+
+* Configure logging to log only http requests and responses including all
+  information in the requests and responses to a file::
+
+    WBEMConnection.configure_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log='all')
+
+Configure pywbem logging using Python logging methods.
+
+* Configure a single pywbem connection with standard Python logger methods by
+  defining the root logger with basicConfig::
+
+    # Set basic logging configuration (i.e root level logger)
+    import logging
+    logging.basicConfig(filename='example.log',level=logging.DEBUG)
+    conn = WBEMConnection(...)
```

Comment:
Did you try this out? This configures the root logger, but we are not propagating to the root logger unless the pywbem loggers are configured.

Reply:
Yes. I have also added a test to confirm it. Everything propagates upward.